src/Chapter3.hs

```diff
@@ -343,6 +344,31 @@ Define the Book product data type. You can take inspiration from our description
 of a book, but you are not limited only by the book properties we described.
 Create your own book type of your dreams!
 -}
+data Date = Date
+    { day   :: Int  -- is there a way to restrict acceptable values?
```

Comment:
That is a very good question!

There are several ways to do that actually.

1. __Smart constructors__. You provide only one way to create the value, through the special function, in which you can make a few validations and checks. You can read a bit more on the pattern here: https://kowainik.github.io/posts/haskell-mini-patterns#smart-constructor 
2. __Custom types__. You can use a custom type to represent that, for example, you can create the sum type for months 🙂 
3. There are very fancy type-level magic tricks, which we will not cover at this point of the course 😄 

If you are interested to know how the types look like in Haskell library, you can check it out here: You can read a bit more on the pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#smart-constructor